corehq/apps/userreports/expressions/specs.py

```diff
@@ -239,3 +240,24 @@ def __call__(self, item, context=None):
         for property_name, expression in self._compiled_properties.items():
             ret[property_name] = expression(item, context)
         return ret
+
+
+class EvalExpressionSpec(JsonObject):
+    type = TypeProperty('evaluator')
+    equation_statement = StringProperty(required=True)
+    variables_expression = DictProperty(required=True)
+
+    def configure(self, variables_expression):
+        self._variables_expression = variables_expression
+
+    def __call__(self, item, context=None):
+        var_dict = self.get_variables(item, context)
+        return eval_statements(self.equation_statement, var_dict)
+
+    def get_variables(self, item, context):
+        var_dict = self._variables_expression(item, context)
```

Comment:
see comment above. right now it looks like you're just assuming that `_variables_expression` will resolve to a `dict` but not validating that. I think it'd be better to just use a normal dict and then you can use the normal expression factory constructor on the values